common/predictive-text/worker/correction/distance-modeler.ts

```diff
@@ -63,12 +63,32 @@ namespace correction {
       // TODO:  Optimize so that we're not frequently recomputing this?
       // TODO:  We might should generalize this so that the probability-to-cost function isn't directly hard-coded.
       //        Seems like a decent first conversion function though, at least.
-      return this.optimalInput.map(mass => mass.p).reduce((previous, current) => previous + (1 - current), 0);
+      //
+      // NOTE:  changes here need to be mirrored in the Node cost function, too!
```

Comment:
Is there a plan to reduce the duplication between, e.g., `inputSamplingCost()` and `currentCost()` in `SearchNode` or `SearchEdge`?

Reply:
You already approved it.  https://github.com/keymanapp/keyman/pull/3565